crates/hdi/src/op.rs

```diff
@@ -18,15 +18,28 @@ mod test;
 /// - Get the [`Op::prev_action()`] of the op.
 /// - Get the [`Op::action_type()`] of the op.
 pub trait OpHelper {
-    /// Converts an [`Op`] to an [`OpType`] without consuming it.
+    /// Converts an [`Op`] to a [`FlatOp`] without consuming it.
     /// This will clone the required internal data.
-    fn to_type<ET, LT>(&self) -> Result<OpType<ET, LT>, WasmError>
+    fn flattened<ET, LT>(&self) -> Result<FlatOp<ET, LT>, WasmError>
```

Comment:
In line with the Rust method `flatten` for Iterators, I suggest

```suggestion
    fn flatten<ET, LT>(&self) -> Result<FlatOp<ET, LT>, WasmError>
```

Reply:
I explicitly wanted to not collide with the name used in Iterator, since this is in a completely different sense

Reply:
Mh, is it? Iterators are flattened from x-dimensional arrays to x-1-dimensional arrays so to say.
Here the multi-depth enums are flattened to one level of depth. Is that completely different?

Reply:
Maybe not *completely* different, but certainly not equivalent. We're modifying the enum *in the sense of* flattening it like nested arrays, but it's not a strict interleaving of one layer into another. We're sometimes bubbling up helpful things from a few layers down, sometimes expanding one variant to two (in the case of Entry and PrivateEntry), but not expanding the sibling variants. It's a very customized flattening operation, unlike the well-defined automatic `flatten`.

Reply:
Ok, I understand.

But then I gotta ask, do you think that `flattened` instead of `flatten` conveys this difference?

I'd rather have this run the risk of being mistaken for a mere flattening of the enum, than using a past tense verb which looks more like a naming negligence than indicating a substantial difference.